### PR TITLE
Added custom builder Gitea example

### PR DESCRIPTION
**Pull request by enrichman**
Fixes: https://github.com/epinio/epinio/issues/2110

### CODE REVIEW

File: docs/howtos/custom_builder_go.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+```
+pack builder create ghcr.io/enrichman/gitea-builder:0.1.0 --config builders/gitea-builder/builder.toml
+```
+
+and push it to a public registry
+
+```
+docker push ghcr.io/enrichman/gitea-builder:0.1.0
+```
```
Comment (andreas-kupries) [inline docs/howtos/custom_builder_go.md:174]: This seems to ne incomplete. The promise of `This How To will show you how to deploy Gitea with Epinio and a custom builder.` does not seem to be fulfilled, as Gitea is not deployed at all. At the stopping point we only have the custom builder.

**Review by andreas-kupries — CHANGES_REQUESTED** [review]

File: docs/howtos/custom_builder_go.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+sidebar_label: "Deploy Gitea with a custom builder image"
+sidebar_position: 14
+title: ""
+---
+
+# Deploy Gitea with a custom builder image
+
+This How To will show you how to deploy Gitea with Epinio and a custom builder.
+
+[Gitea](https://github.com/go-gitea/gitea) is a self-hosted Git service. It's written in Go and Node.js.
```
Comment (andreas-kupries) [inline docs/howtos/custom_builder_go.md:11]:
```suggestion
[Gitea](https://github.com/go-gitea/gitea) is a self-hosted Git service. It is written in Go and Node.js.
```

File: docs/howtos/custom_builder_go.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+At the time of writing the Paketo Go buildpack doesn't support Node.js asset compilation (see [issue #671](https://github.com/paketo-buildpacks/go/issues/671)), so we need to create a custom builder.
+
+This builder will just check the needed Go and Node dependencies, and execute the `make` command to build both the frontend and the backend.
+
+The code used during this guide is available in the [example-builder-gitea](https://github.com/epinio/example-builder-gitea) repository.
+
+
+## Basic Concepts and prerequisites
+
+Before starting it's useful to know some basic concepts around Buildpacks, and have some tools already installed.
```
Comment (andreas-kupries) [inline docs/howtos/custom_builder_go.md:22]:
```suggestion
Before starting it is useful to know some basic concepts around Buildpacks, and have some tools already installed.
```

File: docs/howtos/custom_builder_go.md [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+> Note:  
+> We had to fork and/or reuse the same stack ID in order to reuse some Paketo buildpacks, that are going to work only with these stacks (see https://github.com/paketo-buildpacks/node-engine/blob/main/buildpack.toml#L29).
+
+### Build the stack and publish the image
+
+To build the stack:
+
+```
+./stacks/bionic-full-stack/scripts/create.sh
```
Comment (andreas-kupries) [inline docs/howtos/custom_builder_go.md:39]: Should mention that we are in the main directory of the checkout of the `example-builder-repository` (Well, I am assuming here that this is the working directory.

**Review by andreas-kupries — CHANGES_REQUESTED** [review]

File: docs/howtos/custom_builder_go.md (hunk 4 above)
Comment (enrichman) [reply docs/howtos/custom_builder_go.md:45]: I've added a couple of commands that we were showing the cloning and `cd` into the example repository. I guess it should be enough!

File: docs/howtos/custom_builder_go.md (hunk 1 above)
Comment (enrichman) [reply docs/howtos/custom_builder_go.md:180]: That's fair enough. Added a more complete full example, with more commands showing then how to deploy gitea, and also how to bind a mysql db!

**Review by andreas-kupries — APPROVED** [review]

**Pull request merged**